src/Avalonia.Controls.DataGrid/DataGridCheckBoxColumn.cs

```diff
@@ -192,6 +176,46 @@ protected override object PrepareCellForEdit(IControl editingElement, RoutedEven
                         editingCheckBox.IsChecked = !editingCheckBox.IsChecked;
                     }
                 }
+
+                bool? uneditedValue = editingCheckBox.IsChecked;
+                if(editingEventArgs is PointerPressedEventArgs args)
+                {
+                    void ProcessPointerArgs()
+                    {
+                        // Editing was triggered by a mouse click
+                        Point position = args.GetPosition(editingCheckBox);
+                        Rect rect = new Rect(0, 0, editingCheckBox.Bounds.Width, editingCheckBox.Bounds.Height);
+                        if(rect.Contains(position))
+                        {
+                            EditValue();
+                        }
+                    }
+                    
+                    void OnLayoutUpdated(object sender, EventArgs e)
+                    {
+                        if(!editingCheckBox.Bounds.IsEmpty)
+                        {
+                            editingCheckBox.LayoutUpdated -= OnLayoutUpdated;
+                            ProcessPointerArgs();
+                        }
+                    }
+
+                    if(editingCheckBox.Bounds.IsEmpty)
+                    {
+                        editingCheckBox.LayoutUpdated += OnLayoutUpdated;
```

Comment:
Subscribe on Bounds property changed event instead?
Not sure if it will be more optimal though.

Reply:
I don't think it makes a significant difference and I believe property changed events are slightly less efficient than simple events. There also might be some benefits to delaying the processing to later in the layout pass, which the `LayoutUpdated` event should do.